In [29]:
import pandas as pd


In [91]:
db = pd.read_csv(filepath_or_buffer="filtered_output.tsv", sep="\t")

db

,X.Uploaded_variation,Location,Allele,Gene,Feature,Feature_type,Consequence,cDNA_position,CDS_position,Protein_position,...,phastCons470way_mammalian_rankscore,phyloP100way_vertebrate,phyloP100way_vertebrate_rankscore,phyloP17way_primate,phyloP17way_primate_rankscore,phyloP470way_mammalian,phyloP470way_mammalian_rankscore,popEVE_converted_rankscore,popEVE_pred,popEVE_score
0,19_51965166_CTT/C,19:51965166-51965167,-,ENSG00000256683,ENST00000243644.9,Transcript,frameshift_variant,1528-1529,1286-1287,429,...,-,-,-,-,-,-,-,-,-,-
1,19_51965166_CTT/C,19:51965166-51965167,-,ENSG00000269235,ENST00000595010.5,Transcript,"intron_variant,non_coding_transcript_variant",-,-,-,...,-,-,-,-,-,-,-,-,-,-
2,10_102141320_G/GC,10:102141319-102141320,C,ENSG00000148840,ENST00000278070.7,Transcript,frameshift_variant,2840-2841,2811-2812,937-938,...,-,-,-,-,-,-,-,-,-,-
3,11_113275307_AG/A,11:113275307,-,ENSG00000149294,ENST00000316851.12,Transcript,frameshift_variant,2690,2497,833,...,-,-,-,-,-,-,-,-,-,-
4,11_113275307_AG/A,11:113275307,-,ENSG00000255129,ENST00000526487.6,Transcript,downstream_gene_variant,-,-,-,...,-,-,-,-,-,-,-,-,-,-
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6110,3_52258581_T/C,3:52258581,C,ENSG00000306261,ENST00000816525.1,Transcript,upstream_gene_variant,-,-,-,...,-,-,-,-,-,-,-,-,-,-
6111,3_52258581_T/C,3:52258581,C,ENSG00000306295,ENST00000816808.1,Transcript,downstream_gene_variant,-,-,-,...,-,-,-,-,-,-,-,-,-,-
6112,8_143739878_C/A,8:143739878,A,ENSG00000203499,ENST00000527139.7,Transcript,synonymous_variant,190,105,35,...,-,-,-,-,-,-,-,-,-,-
6113,5_36183911_C/A,5:36183911,A,ENSG00000145604,ENST00000274255.11,Transcript,3_prime_UTR_variant,3307,-,-,...,-,-,-,-,-,-,-,-,-,-


In [92]:
coding_del_order = ["frameshift_variant", "stop_gained", "splice_donor_variant", "splice_acceptor_variant",
                       "inframe_deletion", "inframe_insertion", "missense_variant", "start_lost", "stop_lost", "synonymous_variant"]


cons = list(db["Consequence"])
cons_simple = []
for index, con in enumerate(cons):
    final_string = con
    if "," in con:
        split_cons = con.split(",")
        final = 99
        for val in split_cons:
            if val in coding_del_order:
                rank = coding_del_order.index(val)
                if rank < final:
                    final = rank
        if final != 99:
            final_string = coding_del_order[final]
        else:
            final_string = None
    

    if final_string in coding_del_order:
        cons_simple.append(final_string)
    else:
        cons_simple.append(None)

print(cons_simple)

db['Consequence_simple'] = cons_simple


['frameshift_variant', None, 'frameshift_variant', 'frameshift_variant', None, None, 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', None, 'frameshift_variant', 'frameshift_variant', None, None, 'frameshift_variant', None, 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', None, None, None, 'frameshift_variant', 'frameshift_variant', None, None, 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', None, None, 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', None, 'frameshift_variant', 'frameshift_variant', None, None, 'frameshift_variant', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', None, 'frameshift_variant', 'frameshift_variant', 'frameshift_variant', 'frameshift_v

In [93]:
set(db["AlphaMissense_pred"])



{'-', 'A', 'B', 'LB', 'LP', 'P'}

In [94]:
db.to_csv("cleaned_filtered_output.tsv", index=False, sep="\t")

"chr11:1101310"            "chr8:145065751"          
"chr3:57035120"            "chr11:1102436"
"chr5:36183911" 




chr16:49636754-49636760 (mine)
chr16:49636751-49636757

chr2:232567139-232567140 (mine)
chr2:232567138-232567139

chr16:31224190-31224191 (mine)
chr16:31224189-31224190

1460
3008
3399
3481



